blog for nav and dashboard (#1722)
pulumi/pulumi-hugo: themes/default/content/blog/pulumi-service-updates/index.md

---

> title: "Pulumi Service Updates"

Reviewer: i would go further. maybe like "New and improved Pulumi Service user experience available today"

# The date represents the post's publish date, and by default corresponds with
# the date this file was generated. Posts with future dates are visible in development,
# but excluded from production builds. Use the time and timezone-offset portions of
# of this value to schedule posts for publishing later.
date: 2022-07-14

# Use the meta_desc property to provide a brief summary (one or two sentences)
# of the content of the post, which is useful for targeting search results or social-media
# previews. This field is required or the build will fail the linter test. 
# Max length is 160 characters.

> meta_desc: New user experience for Pulumi Service blah blah blah.

Author: @SaraDPH can you craft a meta description for us?

> meta_desc: New user experience for Pulumi Service blah blah blah.

Author: ```suggestion
meta_desc: Pulumi Service, the easiest way to use Pulumi’s open-source universal infrastructure as code, just got better with a new and improved user experience. Learn all about it.
```

# The meta_image appears in social-media previews and on the blog home page.
# A placeholder image representing the recommended format, dimensions and aspect
# ratio has been provided for you.
meta_image: meta.png

# At least one author is required. The values in this list correspond with the `id`
# properties of the team member files at /data/team/team. Create a file for yourself
# if you don't already have one.
authors:
    - anita-trimbur

# At least one tag is required. Lowercase, hyphen-delimited is recommended.
tags:
    - pulumi-service
    - user-experience

# See the blogging docs at https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md.
# for additional details, and please remove these comments before submitting for review.
---

> Earlier this year, Pulumi's user experience (UX) team sat down with users to understand how we can better serve the community. It's important not only to the UX team, but across the company, that we listen to the real wants, needs, pains, and excitements of the people who use our product.

Reviewer: suggest changing "better serve the community" to something more specific such as "how we can improve Pulumi"

> The [Pulumi Service]({{< relref "/product/pulumi-service" >}}) was at the center of many conversations we had with users. The Pulumi Service is the easiest way to use Pulumi’s open source, universal infrastructure as code SDK at scale and provides a fully managed experience. The team is always exploring how we can deliver the best SaaS experience we can. But having input from the community gave us a meaningful point of entry.

Author: ```suggestion
The [Pulumi Service]({{< relref "/product/pulumi-service" >}}) was at the center of many conversations we had with users. The Pulumi Service is the easiest way to use Pulumi’s open source, universal infrastructure as code SDK at scale and provides a fully managed experience. The team is always exploring the ways in which we can deliver the best Software as a Service (SaaS) experience. Having input from the community gave us a meaningful point of entry.
```

> The [Pulumi Service]({{< relref "/product/pulumi-service" >}}) was at the center of many conversations we had with users. The Pulumi Service is the easiest way to use Pulumi’s open source, universal infrastructure as code SDK at scale and provides a fully managed experience. The team is always exploring how we can deliver the best SaaS experience we can. But having input from the community gave us a meaningful point of entry.

Author: @meagancojocar i have no idea what is going on, on github with this. but i think you suggested these changes on an old version of this file. that's my best guess. can you you resubmit changes you want on the fresh file pls, thank you.

> The [Pulumi Service]({{< relref "/product/pulumi-service/" >}}) was at the center of many conversations we had with users. The Pulumi Service is the easiest way to use Pulumi’s open source, universal infrastructure as code SDK at scale and provides a fully managed experience. The team is always exploring how we can deliver the best SaaS experience we can. But having input from the community gave us a meaningful point of entry.

Author: ```suggestion
The [Pulumi Service]({{< relref "/product/pulumi-service" >}}) was at the center of many conversations we had with users. The Pulumi Service is the easiest way to use Pulumi’s open source, universal infrastructure as code SDK at scale and provides a fully managed experience. The team is always exploring how we can deliver the best SaaS experience we can. But having input from the community gave us a meaningful point of entry.
```

<!--more-->

> Here are the two problems our users faced that we wanted to improve:

Author: ```suggestion
Here are the two problems our users faced that we wanted to address:
```

> Here are the two problems our users faced that we wanted to improve:

Author: @meagancojocar i have no idea what is going on, on github with this. but i think you suggested these changes on an old version of this file. that's my best guess. can you you resubmit changes you want on the fresh file pls, thank you.

> 1. Users were losing time trying to get where they needed to go. Important content needs to be quick and easy to find.

Reviewer: ```suggestion
1. Users need to find important content faster and more easily.
```

> 2. Some users didn't feel like they understood the "big picture" of their operations. We wanted to offer the proper sight lines into an organization's IaC.

Reviewer: can you clarify what this means? it's not clear

We knew we could do better. Now, we're excited to share the changes we've made so far!

## New primary navigation pane

> Next time you sign in to the Pulumi Service, look for the navigation pane on the left hand side of the page. All primary navigation items like your Dashboard, Stacks, and Settings are accessible in the new pane from anywhere in the Service.

Reviewer: "primary navigation items" sounds very UX-y. recommend writing it so it's oriented to the user such as "Now you can easily navigate to common selections like the Dashboard, Stacks, and Settings..."

Reviewer 2: We've been having an open conversation about how to refer to this new navigation in public-facing writing. "Primary navigation pane" is a placeholder I've been using, but we've also talked about "left navigation," "side navigation," "vertical navigation," etc. Do you have any thoughts on the most accessible term to use?

Reviewer 3: "Easily navigate across the Pulumi Service with new Left Navigation"?

![""](dashboard-nav-expanded.png)

> The navigation pane consolidates actions and information so you don't have to look so hard to find where things are. You'll find options in the pane to:

Reviewer: ```suggestion
The navigation pane consolidates actions and information so it's easy for you to find common items. You'll find options in the pane to:
```

- switch between your individual account and any other organizations you're part of
- create new organizations
- access your favorite stacks (read on to find out more!)
- directly access sub-pages like Billing or Access Tokens

> - send us your feedback!

Reviewer: is this a new option?

Author: you could send feedback as part of the get started card, its now available on every page because its part of the navigation panel

![""](dashboard-nav-collapsed.png)

> We made these changes to help users to move through the Pulumi Service without losing their place and ensure constant access to key pages. Even if you collapse the navigation pane for more screen real estate, you can still get where you need to go!

Reviewer: can you clarify what is meant by "even if you collapse the pane... you can still get where you need to go"?

## Stack Favorites

> The Pulumi Service now supports stack favorites. We want to reduce the time practitioners spend combing through projects. Stack favorites offer quick access to the stacks you need most. Look for the star icon next to any stack name and select it to start building out your favorites list.

Reviewer: ```suggestion
You can now pin your favorite stacks, so you can quickly access the stacks you need most. This reduces your time spent combing through projects. Look for the star icon next to any stack name and select it to start building out your favorites list.
```

![""](stack-navigation.png)

> Your favorites list is your own—even if you're part of an organization, your favorites won't be shared with anyone other than you. And you can have as many favorites as you like!

Reviewer: break into separate sentences

You can access the favorites list from two key locations in the Service: the primary navigation pane and the dashboard. All your favorites will appear in the navigation pane under the “Stacks” heading. You'll also have your list of favorites in a new dashboard card.

![""](stacks-navigation-favorites-card.png)

## Pulumi Service dashboard redesign

linter fun _image will be added here_

Lastly, we’ve also made significant changes to the Pulumi Service dashboard. Now you can access more information to help survey your operations at a glance.

Organizations can access some quick usage statistics at the top of their dashboards:

- member count and +/- change in the past week
- total stack count and percent change in the past week
- total resource count and percent change in the past week
- number of successful updates in the past day, week, month, and year

![""](dashboard-quick-stats.png)

Users in organizations will also see a “Members” card that tracks recently joined members. Admins can manage any pending invites from this card.

> All users can now see more of their operation updates in the “Latest Stack Activity” card. You can expose up to ten updates and favorite stacks directly from the dashboard.

Reviewer: there is inconsistent use of first person and third person (users) throughout this blog post. recommend choosing one and making it consistent throughout

![""](members-stack-activity-cards.png)

## Heading? idk

> These changes wouldn’t be possible without the users who generously gave their time to have a conversation with us. We're excited to translate their feedback into a better Pulumi experience. Know that this is the start to even more improvements across the Service and the Pulumi platform. We're always striving to deliver the best product we can.

Reviewer: there should be a call to action in this section to log into Pulumi Service and give it a try

> These changes wouldn’t be possible without the users who generously gave their time to have a conversation with us. We're excited to translate their feedback into a better Pulumi experience. Know that this is the start to even more improvements across the Service and the Pulumi platform. We're always striving to deliver the best product we can.

Reviewer: this paragraph is a little confusing as you are stating that "we're excited to translate their feedback into a better experience" but the blog is saying that you already did that for the UI

---

> title: "Improving the Pulumi Service Experience"

Reviewer: i still think the title can be improved. it sounds a little generic and i think it understates the work you've done.

Reviewer: perhaps hinting at the nature of the improvements might make this more catchy to a reader. i don't think there's anything wrong with "improving" inherently, but i am feeling like the title is non-specific and doesn't have enough draw

Reviewer 3: Yeah - let's do something more exciting here :-).

"A New and Improved User Experience For the Pulumi Service"

"A New Pulumi Service UI: Rich Dashboards, Easier Navigation and Favorites"

(or something else - but maybe more in those directions?)

# Use the meta_desc property to provide a brief summary (one or two sentences)
# of the content of the post, which is useful for targeting search results or social-media
# previews. This field is required or the build will fail the linter test. 
# Max length is 160 characters.
meta_desc: Pulumi Service, the easiest way to use Pulumi’s open-source universal infrastructure as code, just got better with a new and improved user experience.

# The meta_image appears in social-media previews and on the blog home page.
# A placeholder image representing the recommended format, dimensions and aspect
# ratio has been provided for you.
meta_image: meta.png

# At least one author is required. The values in this list correspond with the `id`
# properties of the team member files at /data/team/team. Create a file for yourself
# if you don't already have one.
authors:
    - anita-trimbur

# At least one tag is required. Lowercase, hyphen-delimited is recommended.
tags:
    - pulumi-service
    - user-experience

# See the blogging docs at https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md.
# for additional details, and please remove these comments before submitting for review.
---

> Earlier this year, Pulumi's user experience (UX) team sat down with users to understand how we can improve the Pulumi experience. It's important not only to the UX team, but across the company, that we listen to the real wants, needs, pains, and excitements of the people who use our product.

Reviewer 3: Let's reframe the intro here to focus on the product and how we make it great.  Something closer to this narrative arc:
1. The Pulumi Service enables users and organizations to manage the breadth of cloud infrastructure they use across clouds, with features like X, Y, Z (core messaging from e.g. https://www.pulumi.com/product/pulumi-service/).
2. We initially launched the Pulumi Service console experience in 2018, and although the features and capabilities of the service have grown quickly over the last 4 years, the overall structure of the user experience has stayed largely consistent.
3. Today, we're launching a set of significant improvements to the overall user experience of the Pulumi Service console, making it easier to navigate around the console, surfacing even more important organizational insights into the dashboard and enabling quick access to the stacks you care most about with a new Favorites feature.

Specifically - I would lean less on the internal process we used and more on the product value.  I would also lean less on "here are all the problems with the existing product", and more on "here's what's even better now".

If we do want to have a follow-up blog post focused on UX design and development processes - that might be a good place to talk about some of these other themes - but I'd suggest they distract a bit from the product story as the lede here.

> The [Pulumi Service](https://www.pulumi.com/docs/intro/pulumi-service/) was at the center of many conversations we had with users. The Pulumi Service is the easiest way to use Pulumi’s open source, universal infrastructure as code SDK at scale and provides a fully managed experience. The team is always exploring how we can deliver the best SaaS experience we can. But having input from the community gave us a meaningful point of entry.

Reviewer: ```suggestion
The [Pulumi Service]({{< relref "/product/pulumi-service/" >}}) was at the center of many conversations we had with users. The Pulumi Service is the easiest way to use Pulumi’s open source, universal infrastructure as code SDK at scale and provides a fully managed experience. The team is always exploring how we can deliver the best SaaS experience we can. But having input from the community gave us a meaningful point of entry.
```

<!--more-->

Here are the two problems our users faced that we wanted to improve:

> 1. Users were losing time trying to get where they needed to go. Users need to find important content faster and more easily.

Reviewer: maybe say "wasting time" rather than "losing time"? wasting seems more apt here, but they might just be the same to someone else

Reviewer 2: My hesitation with "wasting" is that it can reflect negatively on the user rather than on the UI. As in, the current UI is what costs users the time, not that the users were wasting their own time.

> 2. Users didn't feel like they understood the "big picture" of their Pulumi usage. We wanted to offer improved sight lines into organizations' day-to-day operations.

Reviewer: should it be "improved line of sight into"?

We knew we could do better. Now, we're excited to share the changes we've made so far!

## New primary navigation pane

> Next time you sign in to the Pulumi Service, look for the navigation pane on the left hand side of the page. All primary navigation items like your Dashboard, Stacks, and Settings are accessible in the new pane from anywhere in the Service.

Reviewer 3: > Next time you sign in to the Pulumi Service

We'll want to rephrase this a little given the decision to launch with both UI's available.  Probably worth a note on the how both UIs are available now, with link to how to try the new UI?

Reviewer 2: Absolutely, I hadn't edited this post since the decision to introduce the toggle. Will be adjusting this today!

![""](dashboard-nav-expanded.png)

The navigation pane consolidates actions and information so it's easy for you to find common items. You'll find options in the pane to:

- switch between your individual account and any other organizations you're part of
- create new organizations
- access your favorite stacks (read on to find out more!)
- directly access sub-pages like Billing or Access Tokens
- send us your feedback!

![""](dashboard-nav-collapsed.png)

> We made these changes to help you move through the Pulumi Service without losing your place and ensure convenient access to key pages. Even if you collapse the navigation pane for more screen real estate, you can still get where you need to go!

Reviewer 3: ```suggestion
We made these changes to help you move through the Pulumi Service without losing your place and to ensure convenient access to key pages. Even if you collapse the navigation pane for more screen real estate, you can still get where you need to go!
```

> If you're looking for your account details, they have moved into the black utility toolbar. To access your account settings or logout of the Pulumi Service, select your profile image at the top right of your browser window.

Reviewer 3: ```suggestion
If you're looking for your user account details, they have moved into the black utility toolbar at the top of the screen. To access your account settings or logout of the Pulumi Service, select your profile image at the top right of your browser window.
```

Perhaps screenshot of this as well?

## Stack Favorites

> You can now pin your favorite stacks, so you can quickly access the stacks you need most. This reduces your time spent combing through projects. Look for the star icon next to any stack name and select it to start building out your favorites list.

Reviewer 3: Might be nice to have a short intro here - along the lines of:

"Most Pulumi users have access to dozens of hundreds of stacks, but regularly interact with a few particularly important stacks as part of their daily usage.  We've introduced a new Stack Favorites feature to let users identify the stacks that are most important to them, and to offer easy access to those stacks from anywhere in the console"

![""](stack-navigation.png)

Your favorites list is your own. Even if you're part of an organization, your favorites won't be shared with anyone other than you. And you can have as many favorites as you like!

You can access the favorites list from two key locations in the Service: the primary navigation pane and the dashboard. All your favorites will appear in the navigation pane under the “Stacks” heading. You'll also have your list of favorites in a new dashboard card.

![""](stacks-navigation-favorites-card.png)

> ## Pulumi Service dashboard redesign

Reviewer 3: ```suggestion
## Richer Insights in the New Pulumi Service Dashboard
```

?

> linter fun _image will be added here_

Reviewer 2: This will be the screenshot of the full dashboard

Lastly, we’ve also made significant changes to the Pulumi Service dashboard. Now you can access more information to help survey your operations at a glance.

Organizations can access some quick usage statistics at the top of their dashboards:

- member count and +/- change in the past week
- total stack count and percent change in the past week
- total resource count and percent change in the past week
- number of successful updates in the past day, week, month, and year

![""](dashboard-quick-stats.png)

If you're part of an organization, you'll also see a “Members” card that tracks recently joined members. If you're an admin, you can manage any pending invites from this card.

We've also exposed more operation updates in the “Latest Stack Activity” card. You can see up to ten most recent updates and favorite those stacks directly from the dashboard card.

![""](members-stack-activity-cards.png)

## Try out the Pulumi Service today

> We'd love for you to take this and improved Pulumi Service for a ride! If you haven't already, you can [create an account](https://app.pulumi.com/signup) and experience these changes for yourself. Don't hesitate to [send us feedback](https://www.pulumi.com/contact/) about any thoughts and feelings you have about these changes or any and all things Pulumi! We're more than happy to learn from our community.

Reviewer: first sentence has grammatical issue

Reviewer: should we plug the community slack as well? i thought that the contact form is typically used for sales queries.

Reviewer: is there also a way to submit GH issues for the Service?
